src/node_api.cc

```diff
@@ -37,9 +38,12 @@ bool node_napi_env__::can_call_into_js() const {
 
 v8::Maybe<bool> node_napi_env__::mark_arraybuffer_as_untransferable(
     v8::Local<v8::ArrayBuffer> ab) const {
-  return ab->SetPrivate(context(),
-                        node_env()->untransferable_object_private_symbol(),
-                        v8::True(isolate));
+  return ab->SetPrivate(
+      context(),
+      node_env()->transfer_mode_private_symbol(),
+      v8::Integer::New(
+          isolate,
+          static_cast<int32_t>(node::BaseObject::TransferMode::kCloneable)));
```

Comment:
Why don't we change the enum to int32_t instead of static casting it everywhere? Is there a specific reason for keeping it as uint8_t?